be/src/olap/rowset/segment_v2/binary_plain_page.h

```diff
@@ -63,9 +63,9 @@ class BinaryPlainPageBuilder : public PageBuilder {
         // If the page is full, should stop adding more items.
         while (!is_page_full() && i < *count) {
             auto src = reinterpret_cast<const Slice*>(vals);
+            _buffer.append(src->data, src->size);
             size_t offset = _buffer.size();
             _offsets.push_back(offset);
-            _buffer.append(src->data, src->size);
```

Comment:
`_offsets` record the start offset of each entry, therefore we should first record offset then append slice data here.